components/TextFields/src/private/MDCPaddedLabel.h

```diff
@@ -0,0 +1,42 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+/**
+ Deltas for padding.
+
+ horizontal is the dx and vertical is the dy.
+ */
+struct MDCSymmetricPadding {
+  CGFloat horizontal;
+  CGFloat vertical;
+};
```

Comment:
Can we not use UIEdgeInsets?
Perhaps with a convenience method: UIEdgeInsetsWithSymetricalPadding(horizontal, vertical)

Reply:
The labels don't have 'real' insets: the left can never be different than the right. And that's by design: it's the only case needed by the class using it and it's a private class.

So I had a conversation with 3 developers asking what would be the most natural way to express these values: CGPoint, CGSize, UIEdgeInsets, Custom struct. After much deliberation, 2:3 vote for custom struct.

Reply:
☹️  Creating a custom struct that is almost, but not quite entirely, identical to an existing, commonly used struct strikes me as unfriendly.
Consider that using a UIEdgeInsets might also simplify your documentation:
/**
 The padding to be applied.

  paddingInsets are used to inset the target rect in drawTextInRect and to increase the intrinsicContentSize.
*/
I'll leave it to you.